gulpfile.js

```diff
@@ -0,0 +1,109 @@
+
+// Properties to make configurable via Twig templates
+const dynamicSvgProps = [
+  'aria-hidden',
+  'aria-labelledby',
+  'class',
+  'focusable',
+  'height',
+  'id',
+  'preserveAspectRatio',
+  'role',
+  'style',
+  'viewBox',
+  'width'
+];
+
+/**
+ * Accepts SVG source markup and templatizes root attributes while also adding
+ * layout blocks (`before`, `content` and `after`) to its contents.
+ *
+ * @param {String} src - The source SVG markup.
+ * @returns {String}
+ */
+function templatizeSvgString(src) {
+  const svg = ltx.parse(src);
+
+  // Create blocks for SVG content, before and after
+  const prepend = ltx.parse(
+    '<root>{% block before %}{% endblock %}{% block content %}</root>'
+  );
+  const append = ltx.parse(
+    '<root>{% endblock %}{% block after %}{% endblock %}</root>'
+  );
+  svg.children = [...prepend.children, ...svg.children, ...append.children];
+
+  // Identify props already in use in the SVG versus those yet to be used
+  const usedProps = dynamicSvgProps.filter(prop => Boolean(svg.attrs[prop]));
+  const unusedProps = dynamicSvgProps.filter(prop => !svg.attrs[prop]);
+
+  // Properties already in use should have their value set to a conditional.
+  // The `default` filter would be less code, but things get tricky when it
+  // comes to managing quotation marks in XML.
+  usedProps.forEach(prop => {
+    const current = svg.attrs[prop];
+    const twigProp = prop.replace(/-/g, '_');
+    svg.attrs[
+      prop
+    ] = `{% if ${twigProp} %}{{${twigProp}}}{% else %}${current}{% endif %}`;
+  });
+
+  // Grab the SVG source to this point
+  let result = svg.root().toString();
+
+  if (unusedProps.length > 0) {
+    // We build a big string of attribute name/value pairs for any properties
+    // yet to be used for this asset.
+    const unusedPropHtml = unusedProps
+      .map(prop => {
+        const twigProp = prop.replace(/-/g, '_');
+        return `{% if ${twigProp} %} ${prop}="{{${twigProp}}}"{% endif %}`;
+      })
+      .join('');
+
+    // We tack this string onto the root SVG element, which we assume ends with
+    // the first occurrence of `>`.
+    result = result.replace('>', `${unusedPropHtml}>`);
+  }
+
+  return result;
+}
+
+/**
+ * Gulp task for converting SVG files to Twig templates.
+ */
+function svgToTwig() {
+  return (
+    src('src/assets/**/*.svg')
+      // Optimize assets with SVGO
+      .pipe(svgmin(svgoConfig))
+      // Pipe file contents through templatize function
+      .pipe(
+        obj((file, _, cb) => {
+          if (file.isBuffer()) {
+            const template = templatizeSvgString(file.contents.toString());
+            file.contents = Buffer.from(template);
+          }
+
+          cb(null, file);
+        })
+      )
+      // Append `.twig` to filenames
+      .pipe(rename({ extname: '.svg.twig' }))
+      // Output to same directory to expose to Storybook
+      .pipe(dest('src/assets'))
```

Comment:
I'm not super familiar with storybook so I don't know if this suggestion is feasible.

Could we move these to `src/assets/compiled` or something, so we have one directory housing all of our gitignored twig files instead of having them intermingled with non-ignored files?

Reply:
@Paul-Hebert It's feasible, but since we need to use relative paths (to keep the templates usable both in the pattern library in our theme), I'm a little sensitive to adding another path segment to all of these that we'll have to type _and_ remember. I'm open to feedback on that, though.

Reply:
Cool, that's fine by me :+1: